### ISSUE TITLE
Feature Request: Microsoft Store Apps/Winget Apps

### ISSUE THREAD

**Issue by SkipToTheEndpoint** [Micke-K/IntuneManagement#101]
While it's very much "hot off the press" and still unavailable on the vast majority of tenants, the new Store for Business replacement of the Microsoft Store App type (https://learn.microsoft.com/en-us/mem/intune/apps/store-apps-microsoft) is now available.

The Graph docs are [here](https://learn.microsoft.com/en-us/graph/api/resources/intune-apps-wingetapp?view=graph-rest-beta), and I've successfully created an app using this app type directly from the Graph Explorer with the following:
POST https://graph.microsoft.com/beta/deviceAppManagement/mobileApps
```
{
  "@odata.type": "#microsoft.graph.winGetApp",
  "categories": [],
  "description": "Microsoft Intune helps organizations manage access to corporate apps, data, and resources. Company Portal is the app that lets you, as an employee of your company, securely access those resources.\r\n\r\nBefore you can use this app, make sure your IT admin has set up your work account. Your company must also have a subscription to Microsoft Intune.\r\n\r\nCompany Portal helps simplify the tasks you need to do for work:\r\n\r\n- Enroll your device to access corporate resources, including Office, email, and OneDrive for Business\r\n- Sign in to corporate resources with company-issued certificates\r\n- Reduce the number of times you need to sign in to corporate resources with single sign-on\r\n- Browse and install approved business apps from your IT department or the Microsoft Store for Business\r\n- View and manage your enrolled devices – and wipe them if they get lost or stolen\r\n- Get help directly from your IT department through available contact information\r\n \r\nA note about Intune: every organization has different access requirements, and will use Intune in ways that they determine will best manage their information. Some functionality might be unavailable in certain countries. If you have questions about how this app is being used within your organization, your company’s IT administrator should have those answers for you. Microsoft, your network provider, and your device’s manufacturer do not know how Intune will be used by your organization.",
  "developer": "",
  "displayName": "Company Portal",
  "informationUrl": "",
  "isFeatured": false,
  "roleScopeTagIds": [],
  "notes": "",
  "owner": "",
  "privacyInformationUrl": "http://go.microsoft.com/fwlink/?LinkID=316999",
  "publisher": "Microsoft Corporation",
  "packageIdentifier": "9WZDNCRFJ3PZ",
  "repositoryType": "microsoftStore",
  "manifestHash": "wingetv1:EiVo3jBjKACw3jmkKgszOpFjukAU5qL4m66hL2Vt7qQ=",
  "installExperience": {
    "runAsAccount": "user"
  }
}
```

Having tried the current version of the IntuneManagement tool, it does seem to back up the json from a tenant, albeit without the manifestHash property, as well as successfully import to a flighted tenant (you get a "Windows Store repository apps feature is not supported for this tenant." response otherwise). I haven't yet tested whether the imported app functions as expected, and it seems to log using the `Start-PreImportCommandApplication` function Write-Log on line 1954.

As always happy to help testing on an accessible tenant where possible.

**Comment by Micke-K** [comment]
Hello!

This is interesting! I haven't looked at it myself but I'll sure do as soon as I get the chance. 

I've worked a lot on the documentation request. I had to completely rewrite some parts of the documentation feature but  I'm almost done now. I'll upload that code to the other thread as soon as I finished testing so you can test as well. I'm a bit busy this week but I'll see what I can do. 

Cheers!

**Comment by SkipToTheEndpoint** [comment]
Just to feed back that I've done some testing now I have another tenant with the feature enabled.
I can confirm that an export of the json from an existing tenant (which doesn't include the manifestHash property) successfully imports into a different tenant. Assigning the application to devices successfully delivers the application with no issues.

Seems your code supports them already, though the logging is a bit confused. :)

**Comment by dotjesper** [comment]
Talking about the various app types, and now with the New Store Experience, where apps can be from various places, a Type column would be useful :)

I could properly argue, this extra column would be useful for Policies as well.

**Comment by SkipToTheEndpoint** [comment]
> a Type column

I think the UI cheats. The main "Apps" page has a "Type" column, but I think it's just using a fixed backend reference to the Graph OData type (#microsoft.graph.win32LobApp, #microsoft.graph.officeSuiteApp etc.) as there doesn't seem to be an App Type property when hitting the /deviceAppManagement/mobileApps API endpoint.

The fact that the new apps can be either UWP or Win32 is again not pulled from the Graph, as far as I can see, and is actually held within the app manifest, which sits on an entirely different API.

**Comment by Micke-K** [comment]
Hello

All this should be included in 3.8.1

You can add the column with this configuration

![image](https://user-images.githubusercontent.com/43226266/214826060-ac8e6489-64cc-45c3-ae65-411d8bc2b199.png)

Let me now if you test it or if I can now close this issue

Cheers!
